src/js/tetris.js

```diff
@@ -0,0 +1,235 @@
+import I from './i';
+import J from './j';
+import L from './l';
+import O from './o';
+import S from './s';
+import T from './t';
+import Z from './z';
+import Block from './block';
+import { CONSTANTS } from './values';
+
+const SHAPES = [I, J, L, O, S, T, Z];
+const KEYS = {
+  left: 37,
+  right: 39,
+  down: 40,
+  r: 82,
+  space: 32
+};
+function sleep(time) {
+  return new Promise(resolve => setTimeout(resolve, time));
+}
+
+export default class Tetris {
+  constructor({ container, unitSize }) {
+    this.container = container;
+    this.unitSize = unitSize;
+    this.blocks = [];
+    this.round = 1;
+    this.pause = false;
+    this.speed = 1000;
+    this.score = 0;
+    this.isRunning = false;
+    this.highScore = 0;
+  }
+
+  get elements() {
+    return {
+      score: document.querySelector('.score__value'),
+      highScore: document.querySelector('.high-score__value'),
+      nextShape: document.querySelector('.next-shape__object'),
+      pause: document.querySelector('.pause'),
+      tetris: document.querySelector('.tetris'),
+      gameOver: document.querySelector('.game-over')
+    };
+  }
+
+  start() {
+    this.isRunning = true;
+    if (!this.gameOver) {
+      this.setupListeners();
```

Comment:
We always want to `setupListeners` when we start the game. The gameOver logic can be separated. Let's make a `gameOver` function that gets called when the player looses. And this game over function can remove the listeners added here.